File: bot.js
const { Telegraf, Markup } = require('telegraf');

const bot = new Telegraf('TYT_TOKEN');
const ADMIN_CHAT = 'TYT_ID_CHAT_ADMIN';

const userRequests = {};
const messageCounter = {};

const getAccountAge = (userId) => {
    const timestamp = (userId >> 32) * 1000;
    if (timestamp < 1000000000000) return 'неизвестно';
    const date = new Date(timestamp);
    return date.toLocaleDateString('ru-RU');
};

bot.start(async (ctx) => {
    const user = ctx.from;
    const userId = user.id;
    const accountAge = getAccountAge(userId);
    
    const welcomeMessage = `🎉 Добро пожаловать в поддержку Zenettany!\n\n` +
        `📋 Ваша информация:\n` +
        `👤 Никнейм: ${user.first_name}${user.last_name ? ' ' + user.last_name : ''}\n` +
        `🔖 ИД аккаунта: <code>${userId}</code>\n` +
        `📅 Дата создания аккаунта: ${accountAge}\n` +
        `🌐 Username: @${user.username || 'не установлен'}\n\n` +
        `💎 Разработчики:\n` +
        `• Zenettany: @kannadec\n` +
        `• Gromova\n\n` +
        `• aurproject.eu\n\n` +
        `🔧 Для обращения: /help [ваш вопрос]`;
    
    const keyboard = Markup.inlineKeyboard([
        [Markup.button.callback('Открыть тг Gromova', 'gromova_tg')],
        [Markup.button.url('🌐 AUR PROJECT', 'https://aurproject.eu')],
        [Markup.button.url('📢 Наш канал', 'https://t.me/kannadec_dev')]
    ]);
    
    await ctx.reply(welcomeMessage, {
        parse_mode: 'HTML',
        ...keyboard
    });
});

bot.action('gromova_tg', (ctx) => {
    ctx.answerCbQuery();
    ctx.reply('Не работает? Значит не судьба, бро... Всё что нужно — уже здесь 😉');
});

bot.command('help', (ctx) => {
    const args = ctx.message.text.split(' ');
    if (args.length < 2) {
        return ctx.reply('Напишите /help и ваш вопрос');
    }
    
    const question = args.slice(1).join(' ');
    const user = ctx.from;
    const userKey = `${user.id}_${ctx.chat.id}`;
    
    messageCounter[userKey] = (messageCounter[userKey] || 0) + 1;
    const shortId = `${user.id}_${messageCounter[userKey]}`;
    
    userRequests[shortId] = {
        id: user.id,
        username: user.username,
        chatId: ctx.chat.id
    };
    
    const adminMessage = `🆕 НОВЫЙ ЗАПРОС\n` +
        `👤 ID: <code>${user.id}</code>\n` +
        `📛 Юзернейм: @${user.username || 'нет'}\n` +
        `🆔 ID для ответа: <code>${shortId}</code>\n` +
        `📝 Текст: ${question}\n\n` +
        `📤 Ответить: /otvet ${user.id} [текст]`;
    
    ctx.telegram.sendMessage(ADMIN_CHAT, adminMessage, { parse_mode: 'HTML' });
    ctx.reply('✅ Запрос отправлен администраторам.');
});

bot.command('otvet', (ctx) => {
    if (String(ctx.chat.id) !== ADMIN_CHAT) return;
    
    const args = ctx.message.text.split(' ');
    if (args.length < 3) {
        return ctx.reply('Используйте: /otvet [ID_пользователя] [текст ответа]');
    }
    
    const targetId = args[1];
    const reply = args.slice(2).join(' ');
    
    let userId = null;
    let username = null;
    
    if (targetId.startsWith('@')) {
        const searchUsername = targetId.substring(1);
        for (const [shortId, data] of Object.entries(userRequests)) {
            if (data.username && data.username.toLowerCase() === searchUsername.toLowerCase()) {
                userId = data.id;
                username = data.username;
                break;
            }
        }
    } else {
        const numericId = parseInt(targetId);
        for (const [shortId, data] of Object.entries(userRequests)) {
            if (data.id === numericId) {
                userId = data.id;
                username = data.username;
                break;
            }
        }
    }
    
    if (!userId) {
        return ctx.reply(`❌ Пользователь с ID/юзернеймом "${targetId}" не найден в активных запросах.`);
    }
    
    ctx.telegram.sendMessage(userId, `💌 Ответ от поддержки Zenettany:\n\n${reply}\n\nРазработчик: @kannadec | aurproject.eu`)
        .then(() => ctx.reply(`✅ Ответ отправлен пользователю:\nID: <code>${userId}</code>\nЮзернейм: @${username || 'нет'}`, { parse_mode: 'HTML' }))
        .catch((err) => ctx.reply(`❌ Ошибка: ${err.message}`));
});

bot.on('text', (ctx) => {
    if (ctx.chat.type === 'private' && !ctx.message.text.startsWith('/')) {
        ctx.reply('Для обращения используйте команду /help [ваш вопрос]');
    }
});

bot.launch().then(() => {
    console.log('✅ Бот запущен: AUR PROJECT Support');
});

process.once('SIGINT', () => bot.stop('SIGINT'));
process.once('SIGTERM', () => bot.stop('SIGTERM'));

File: bot.py
import asyncio
import logging
import signal
import sys
from aiogram import Bot, Dispatcher, types, F
from aiogram.filters import Command
from aiogram.types import Message
from aiogram.client.default import DefaultBotProperties

logging.basicConfig(level=logging.INFO, format='%(asctime)s - %(name)s - %(levelname)s - %(message)s')
logger = logging.getLogger(__name__)

bot = Bot(
    token="TYT_TOKEN",
    default=DefaultBotProperties(parse_mode="HTML")
)
dp = Dispatcher()
ADMIN_CHAT = "TYT_ID_CHAT_ADMIN"
user_requests = {}
message_counter = {}

@dp.message(Command("start"))
async def start_cmd(message: Message):
    await message.answer("✅ Бот поддержки Zenettany активен. Используйте /help для обращения.")

@dp.message(Command("help"))
async def help_cmd(message: Message):
    if len(message.text.split()) < 2:
        await message.answer("Используйте: /help [ваш вопрос]")
        return
    
    request_text = message.text.split(maxsplit=1)[1]
    request_id = message.message_id
    
    user_key = f"{message.from_user.id}_{message.chat.id}"
    if user_key not in message_counter:
        message_counter[user_key] = 1
    else:
        message_counter[user_key] += 1
    
    short_id = f"{message.from_user.id}_{message_counter[user_key]}"
    
    user_requests[short_id] = {
        'user_id': message.from_user.id,
        'username': message.from_user.username,
        'full_message_id': request_id
    }
    
    admin_msg = (
        f"🆕 НОВЫЙ ЗАПРОС\n"
        f"👤 ID: <code>{message.from_user.id}</code>\n"
        f"📛 Username: @{message.from_user.username or 'Нет'}\n"
        f"🆔 Короткий ID: <code>{short_id}</code>\n"
        f"📝 Текст: {request_text}\n\n"
        f"📤 Для ответа: /otvet {short_id} [текст]\n"
        f"📤 Или: /otvet @{message.from_user.username} [текст]"
    )
    await bot.send_message(ADMIN_CHAT, admin_msg)
    await message.answer("✅ Ваш запрос отправлен администраторам. Ожидайте ответа.")

@dp.message(Command("otvet"))
async def reply_cmd(message: Message):
    if str(message.chat.id) != ADMIN_CHAT:
        return
    
    args = message.text.split(maxsplit=2)
    if len(args) < 3:
        await message.answer("Используйте: /otvet [короткий_id/username] [текст ответа]")
        await message.answer("Примеры:\n/otvet 123456_1 Привет\n/otvet @username Привет")
        return
    
    target = args[1]
    reply_text = args[2]
    
    user_id = None
    username = None
    
    if target.startswith('@'):
        username = target[1:].lower()
        for short_id, user_data in user_requests.items():
            if user_data['username'] and user_data['username'].lower() == username:
                user_id = user_data['user_id']
                username = user_data['username']
                break
    else:
        if target in user_requests:
            user_data = user_requests[target]
            user_id = user_data['user_id']
            username = user_data['username']
    
    if not user_id:
        await message.answer(f"❌ Пользователь не найден: {target}")
        return
    
    try:
        await bot.send_message(
            user_id,
            f"💌 Ответ от поддержки:\n\n{reply_text}\n\nРазработчик: @kannadec | aurproject.eu"
        )
        
        await message.answer(
            f"✅ Ответ отправлен пользователю:\n"
            f"👤 ID: <code>{user_id}</code>\n"
            f"📛 @{username or 'Нет'}\n"
            f"📝 Ответ: {reply_text[:50]}..."
        )
        
    except Exception as e:
        await message.answer(f"❌ Ошибка при отправке: {str(e)}")

@dp.message(F.text)
async def handle_text(message: Message):
    if message.chat.type == "private" and not message.text.startswith("/"):
        await message.answer("Для обращения используйте команду /help [ваш вопрос]")

async def on_shutdown():
    logger.info("Бот выключается...")
    await bot.session.close()

async def main():
    dp.shutdown.register(on_shutdown)
    
    try:
        logger.info("Бот запускается...")
        await dp.start_polling(bot, skip_updates=True)
    except KeyboardInterrupt:
        logger.info("Бот остановлен пользователем")
    except Exception as e:
        logger.error(f"Ошибка: {e}")
    finally:
        await on_shutdown()

def signal_handler(signum, frame):
    logger.info(f"Получен сигнал {signum}, завершаю работу...")
    sys.exit(0)

if __name__ == "__main__":
    signal.signal(signal.SIGINT, signal_handler)
    signal.signal(signal.SIGTERM, signal_handler)
    
    try:
        asyncio.run(main())
    except KeyboardInterrupt:
        logger.info("Работа завершена")
    except Exception as e:
        logger.error(f"Критическая ошибка: {e}")